### PR TITLE
Fix #53513 - parse error in manage.versions

### LINKED ISSUE #53513: salt-run manage.versions produces ValueError exception
### Description of Issue
salt-run manage.versions produces ValueError exception

```python
# salt-run manage.versions
Exception occurred in runner manage.versions: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 387, in _low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/runners/manage.py", line 697, in versions
    minion_version = salt.version.SaltStackVersion.parse(minions[minion])
  File "/usr/lib/python2.7/site-packages/salt/version.py", line 269, in parse
    'Unable to parse version string: \'{0}\''.format(version_string)
ValueError: Unable to parse version string: 'ERROR executing 'test.version': Executor 'direct_call.get' is not available'
```

### Setup
Salt-master and 2 minions.

### Steps to Reproduce Issue
Not quite sure. I think one of the minions was in the middle of being updated when `salt-run manage.versions` was ran.

### Versions Report
```yaml
Salt Version:
           Salt: 2018.3.3

Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.26.3
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.4
         Python: 2.7.5 (default, Jul 13 2018, 13:06:57)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.5.1804 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-862.11.6.el7.x86_64
         system: Linux
        version: CentOS Linux 7.5.1804 Core
```

### PR DESCRIPTION
**Pull request by waynew**
### What does this PR do?

This changes manage.versions naive approach to calling minion `test.version`,
which always assumed the response would be a version number. Now we only assume
it's a version number if `test.version` returns 0 retcode. Otherwise we will
say that the minion is offline.

### What issues does this PR fix or reference?
Fixes: #53513

### Previous Behavior

When calling `salt-run manage.versions` it would show a stack trace if there
was a failure for the minion to respond (e.g. the minion was in a blackout, or
some other reason). Now it will report that the minion is offline instead. We
*could* add a new "offline" status to the results, but this was the least
invasive change.

### New Behavior

Now `salt-run manage.versions` will report the minion offline when it returns a
non-zero return code.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/integration/runners/test_manage.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import pytest
+
+
+@pytest.fixture
+def blackout_minion(salt_minion, salt_master, salt_run_cli):
+    with pytest.helpers.temp_file(
+        "top.sls",
+        """base:\n  '*':\n    - fnord""",
+        salt_master.pillar_tree.base.paths[0],
```
Comment (s0undt3ch) [inline tests/pytests/integration/runners/test_manage.py:9]:
```suggestion
        salt_master.pillar_tree.base.paths.write_path,
```
Should do the same.

Comment (s0undt3ch) [reply tests/pytests/integration/runners/test_manage.py:9]: Actually, use `salt_master.pillar_tree.base.temp_file`

**Review by Ch3LL — DISMISSED** [review]
approved, pending pedro's suggested changes.

File: tests/pytests/integration/runners/test_manage.py (hunk 1 above)
Comment (waynew) [reply tests/pytests/integration/runners/test_manage.py]: Done. Hopefully tests pass tonight :joy:

**Comment by waynew** [comment]
welp, tests did not all pass - pretty sure just flaky tests. Kicked off a re-run, let's see if that works.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**